frontend/src/metabase/core/components/Button/Button.tsx

```diff
@@ -27,7 +27,7 @@ const BUTTON_VARIANTS = [
   "fullWidth",
 ] as const;
 
-interface Props extends ButtonHTMLAttributes<HTMLButtonElement> {
+interface Props extends React.ComponentProps<typeof ButtonRoot> {
```

Comment:
ButtonHTMLAttributes<HTMLButtonElement> doesn't have a property `as` which I used in https://github.com/metabase/metabase/pull/22483/files#diff-52671715e9e6f09457022a4e9730e2a82cd66aa3c3bff74051e212345104fcf6R72